`make run-ios` fails with a false positive error randomly

# Problem

As reported by @FFFra and @shivekkhurana 
`make run-ios` failed with this error message : 

```
Building status-go for: iossimulator/amd64
buildPhase completed in 1 minutes 3 seconds
installing
post-installation fixup
checking for references to /private/tmp/nix-build-status-go-0.171.39-debfe1c-ios.drv-0/ in
 /nix/store/qggg29bsygh6r460c9a6kj96wjacscq9-status-go-0.171.39-debfe1c-ios...
patching script interpreter paths in 
/nix/store/qggg29bsygh6r460c9a6kj96wjacscq9-status-go-0.171.39-debfe1c-ios
Copying Statusgo.xcframework from Nix store to 
/Users/fraschetti/Status/status-mobile/modules/react-native-status/ios/RCTStatus
No changes detected.
Done in: 1.202s
Checking for modifications in ios/Pods...
No modifications detected.
warn Package react-native-blob-util contains invalid configuration: "dependency.hooks" is not allowed. 
Please verify it's properly linked using "react-native config" command and contact the package maintainers about 
this.warn Package react-native-navigation contains invalid configuration: "dependency.assets" is not allowed,
"dependency.hooks" is not allowed. Please verify it's properly linked using "react-native config" command and 
contact the package maintainers about this.
[fmt] Compiling fmt-dummy.m
[RCT-Folly] Compiling Assume.cpp
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch 
block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason 
"Error: Failed to build iOS project. We ran "xcodebuild" command but it exited with error code 65. 
To debug build logs further, consider building your app with Xcode.app, by opening StatusIm.xcworkspace.".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
```

Screenshot by Shivek:

<img width="1792" alt="Screenshot_2024-01-10_at_10 32 42_AM" src="https://github.com/status-im/status-mobile/assets/64726664/e39ed63b-ce19-4472-b6f9-09e48907a878">

However the iOS build works fine when we do this : 
- `make shell TARGET=ios`
- `npx react-native run-ios`

This means there is some issue in run-ios make command when we attach the status-go target.
The source of make run-ios is this : 

```
SIMULATOR=iPhone 13
run-ios: export TARGET := ios
run-ios: export IOS_STATUS_GO_TARGETS := iossimulator/amd64
run-ios: ##@run Build iOS app and start it in a simulator/device
ifneq ("$(SIMULATOR)", "")
	npx react-native run-ios --simulator="$(SIMULATOR)" | xcbeautify
else
	npx react-native run-ios | xcbeautify
endif
```

The issue could be potentially here : `run-ios: export IOS_STATUS_GO_TARGETS := iossimulator/amd64` cc @yakimant

I can indeed reproduce this on `macm2-01.ih-eu-mda1.ci.devel`:
```
warn Package react-native-blob-util contains invalid configuration: "dependency.hooks" is not allowed. Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.
warn Package react-native-navigation contains invalid configuration: "dependency.assets" is not allowed,"dependency.hooks" is not allowed. Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "Error: No simulator available with name "iPhone 13"".] {
  code: 'ERR_UNHANDLED_REJECTION'
}

Node.js v18.16.0
make: *** [run-ios] Error 1
```
And setting `IOS_STATUS_GO_TARGETS=ios/arm64` or other values doesn't seem to have an effect.

--- a/Makefile
+++ b/Makefile
@@ -279,8 +279,9 @@ run-android: ##@run Build Android APK and start it on the device
 	npx react-native run-android --appIdSuffix debug
 
 SIMULATOR=iPhone 13
+# TODO: fix IOS_STATUS_GO_TARGETS to be either amd64 or arm64 when RN is upgraded
 run-ios: export TARGET := ios
-run-ios: export IOS_STATUS_GO_TARGETS := iossimulator/amd64
+run-ios: export IOS_STATUS_GO_TARGETS := ios/arm64,iossimulator/amd64
 run-ios: ##@run Build iOS app and start it in a simulator/device
 ifneq ("$(SIMULATOR)", "")
 	npx react-native run-ios --simulator="$(SIMULATOR)" | xcbeautify
@@ -291,8 +292,9 @@ endif
 show-ios-devices: ##@other shows connected ios device and its name
 	xcrun xctrace list devices
 
+# TODO: fix IOS_STATUS_GO_TARGETS to be either amd64 or arm64 when RN is upgraded
 run-ios-device: export TARGET := ios
-run-ios-device: export IOS_STATUS_GO_TARGETS := ios/arm64
+run-ios-device: export IOS_STATUS_GO_TARGETS := ios/arm64,iossimulator/amd64
 run-ios-device: ##@run iOS app and start it on a connected device by its name
 ifndef DEVICE_NAME
 	$(error Usage: make run-ios-device DEVICE_NAME=your-device-name)

